python/cudf/bindings/csv.pyx

```diff
@@ -298,3 +298,72 @@ cpdef cpp_read_csv(
     nvtx_range_pop()
 
     return df
+
+cpdef cpp_write_csv(
+    cols, path=None,
+    sep=',', na_rep='',
+    columns=None, header=True, line_terminator='\n'):
+    """
+    Cython function to call into libcudf API, see `write_csv`.
+
+    See Also
+    --------
+    cudf.io.csv.write_csv
+    """
+
+    nvtx_range_push("CUDF_WRITE_CSV", "purple")
+
+    cdef csv_write_arg csv_writer = csv_write_arg()
+
+    path = str(os.path.expanduser(str(path))).encode()
+    csv_writer.filepath = path
+    line_terminator = line_terminator.encode()
+    csv_writer.line_terminator = line_terminator
+    csv_writer.delimiter = sep.encode()[0]
+    na_rep = na_rep.encode()
+    csv_writer.na_rep = na_rep
+    # Do not expose true_value and false_value until gdf_bool type
+    # changes added to cpp API
+    true_value = 'true'.encode()
+    csv_writer.true_value = true_value
+    false_value = 'false'.encode()
+    csv_writer.false_value = false_value
+    csv_writer.include_header = header
+
+    cdef vector[gdf_column*] list_cols
+
+    if columns is not None:
+        if not isinstance(columns, list):
+            raise TypeError('columns must be a list')
+        for name in columns:
+            if name not in cols:
+                raise NameError('column {!r} does not exist in DataFrame'
+                                .format(name))
+            check_gdf_compatibility(cols[name])
+            c_col = column_view_from_column(cols[name]._column)
+            name = str(name).encode()
+            c_col.col_name = name
```

Comment:
We may want to consider pushing this functionality into an optional parameter in `column_view_from_column`, where if name is None it sets it to `nullptr`, otherwise sets it to the name.

Reply:
And ideally would we want this functionality to lie in the Cython layer, or pass it down as a parameter to `gdf_column_view_augmented` where we set the column name?

Reply:
I'm not sure on the implications of adding it to the `gdf_column_view_augmented` function in libcudf, so my gut would be to add it to Cython for now and we can open an issue to discuss libcudf changes and adjust later.

Reply:
There's an interesting bug with the current approach I have here. I set `col.col_name = name` and push this `c_col` into the vector each iteration.. after the loop is done when I check the column names for all elements in the vector, the names I'm left with are `"col3","col3","col3"` instead of the expected `"col1","col2","col3"`.
My theory(might be wrong): Python uses the same memory location to store the `name` in each iteration of the loop. `c_col.col_name` stores the reference to this variable `name`, so at the end all my elements of the vector effectively point to the same `name`.

I haven't pushed my changes for moving this logic to `column_view_from_column` but I was facing a similar issue in that case as well.

Reply:
Correct, it just has a pointer to the Python string object as opposed to allocating and managing its own memory, so those Python objects need to not get garbage collected.